go/vt/tabletserver/planbuilder/dml.go

```diff
@@ -29,6 +29,11 @@ func analyzeUpdate(upd *sqlparser.Update, getTable TableGetter) (plan *ExecPlan,
 		return nil, err
 	}
 
+	// Store the WHERE clause as string for the transaction serializer.
+	buf := sqlparser.NewTrackedBuffer(nil)
+	buf.Myprintf("%v", upd.Where)
```

Comment:
Looks very good and much simpler. Thanks! :)

The semaphore basically allowed you to fold the two layers `TxSerializer` and `queue` into one and that makes the locking much easier to reason about. Additionally, it provides a lot of thread safe functionality for free which must not be implemented again.

Let me look into the fairness of this tomorrow and then I'll backport it into this PR. :)